Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -87,11 +87,39 @@ std::vector<std::string> splitSmartsIntoComponents(
   }
   return res;
 }
+
+ROMol* constructMolFromString(const std::string &txt,
+    std::map<std::string,std::string> *replacements, bool useSmiles, bool sanitize) {
+  ROMol *mol;
+  if(!useSmiles){
+    mol=SmartsToMol(txt,0,false,replacements);
+    if(sanitize){
+      try{
+        mol->updatePropertyCache();
+        RingInfo *ringInfo=mol->getRingInfo();
+        if(!ringInfo->isInitialized()){
+          ringInfo->initialize();
```

Comment:
This initializes the `RingInfo` structure but doesn't put anything in it. So it looks like the molecule has no rings. Is that what you intended or do you mean to be calling one of the ring finding functions instead of the code from line 99 onwards?